src/api/controller/basecontroller.ts

```diff
@@ -0,0 +1,48 @@
+import { Response } from 'express';
+import { RateLimitRequestHandler, rateLimit } from 'express-rate-limit';
+
+import logger from '../../helper/logger';
+
+export default class BaseController {
+	public log: typeof logger;
+
+	public limiter: RateLimitRequestHandler;
+
+	constructor(limitTime = 15) {
+		this.log = logger;
+
+		this.limiter = rateLimit({
+			windowMs: limitTime * 60 * 1000,
+			max: 100,
+		});
+	}
+
+	sendResponse(res: Response, data: any, status = 200) {
+		return res.status(status).send({
+			data,
+		});
+	}
+
+	handleError(res: Response, error: any, code = 400, renderErrorPage = false) {
+		let statusCode: number;
+
+		if (typeof error === 'object' && error.statusCode) {
+			statusCode = error.statusCode;
+		} else {
+			statusCode = code;
+		}
+
+		this.log.error(error);
+
+		if (renderErrorPage) {
+			res.status(statusCode).render(code === 400 ? '404' : code.toString(), {
+				statusCode,
+				error,
+			});
+		} else {
+			res.status(statusCode).send({
+				error,
+			});
```

Comment:
```suggestion
			return res.status(statusCode).render(code === 400 ? '404' : code.toString(), {
				statusCode,
				error,
			});
		} else {
			return res.status(statusCode).send({
				error,
			});
```

Any specific reason not to return the response here? Technically the `.send` method does terminate the request lifecycle already, but this ensures the function manually exits instead of needing to check the call stack again.

Reply:
Actually good point. I've never thought about that before since I'm used to just not return when the last statement terminates.

Reply:
It's a convention I've grown fond of using over time to indicate that the operation being returned is in fact the last operation that needs to be processed.

Without a hard return, it makes it seem like some additional logging/metrics will be captured once the error response is returned, but in this case, it is not.

I'd be happy to make a PR to add it to any of the necessary places if this pattern works for you.